src/main/java/org/folio/linked/data/integration/kafka/consumer/DataImportEventHandler.java

```diff
@@ -24,19 +24,31 @@ public class DataImportEventHandler {
 
   public void handle(DataImportEvent event) {
     if (isNotEmpty(event.getMarcBib())) {
-      var marc4ldResource = marcBib2ldMapper.fromMarcJson(event.getMarcBib());
-      var id = resourceService.createResource(marc4ldResource);
-      log.info("DataImportEvent with id [{}] was saved as LD resource with id [{}]", event.getId(), id);
+      createBibResource(event);
     } else if (isNotEmpty(event.getMarcAuthority())) {
-      var authorityResources = marcAuthority2ldMapper.fromMarcJson(event.getMarcAuthority());
-      var ids = authorityResources.stream()
-        .map(resourceService::createResource)
-        .toList();
-      log.info("Processing MARC Authority record with event ID [{}], saved records: {}",
-        event.getId(), ids.size());
+      createAuthorities(event);
     } else {
       log.error("DataImportEvent with id [{}], tenant [{}], eventType [{}] has no Marc record inside",
         event.getId(), event.getTenant(), event.getEventType());
     }
   }
+
+  private void createBibResource(DataImportEvent event) {
+    marcBib2ldMapper.fromMarcJson(event.getMarcBib())
+      .ifPresentOrElse(resource -> {
+          var id = resourceService.createResource(resource);
+          log.info("DataImportEvent with id [{}] was saved as LD resource with id [{}]", event.getId(), id);
+        },
+        () -> log.warn("Empty resource from MARC for event ID [{}]", event.getId())
```

Comment:
minor
maybe **Unsupported** instead of **Empty**?

Reply:
I think Empty better describes the use case, we're trying to convert resource from MARC here and may receive Optional.empty() not only because unsupported type of resource but for whatever reason.